src/tqec/templates/constructions/qubit.py

```diff
@@ -1,95 +1,120 @@
         for source, relpos, target in _relations:
             self.add_relation(source, relpos, target)
 
 
-class ScalableQubitRectangle(TemplateOrchestrator):
+class QubitRectangleTemplate(ComposedTemplate):
     def __init__(
-        self, k_width: int, k_height: int, scale_width: bool | None = None
+        self,
+        k_width: int | Dimension,
+        k_height: int | Dimension,
+        scale_width: bool | None = None,
     ) -> None:
         """A scalable rectangle error-corrected qubit.
 
-        A scalable rectangle qubit can only scale its width **or** height, but not both.
-        ```text
-        .  .  1  .  1  .  1  .
-        2  3  4  3  4  3  4  .
-        .  4  3  4  3  4  3  5
-        2  3  4  3  4  3  4  .
-        .  4  3  4  3  4  3  5
-        .  6  .  6  .  6  .  .
-        ```
-
-        :param k_width: half the width of the qubit.
-        :param k_height: half the height of the qubit.
-        :param scale_width: whether to scale the width or height. If None, the dimension
-            with the larger value will be scaled. If both dimensions are equal, the width
-            will be scaled by default.
+        TODO
         """
-        width, height = 2 * k_width, 2 * k_height
+        swidth, sheight = self.get_dimensions(k_width, k_height, scale_width)
+        # nsone: non-scalable one
+        nsone = Dimension(1, is_fixed=True)
+
         _templates = [
             # Central square, containing plaquettes of types 3 and 4
-            TemplateWithIndices(ScalableRectangle(width, height, scale_width), [3, 4]),
+            TemplateWithIndices(AlternatingRectangleTemplate(swidth, sheight), [3, 4]),
             # Top rectangle, containing plaquettes of type 1 only
-            TemplateWithIndices(ScalableRectangle(width, 1), [0, 1]),
+            TemplateWithIndices(AlternatingRectangleTemplate(swidth, nsone), [0, 1]),
             # Left rectangle, containing plaquettes of type 2 only
-            TemplateWithIndices(ScalableRectangle(1, height), [2, 0]),
+            TemplateWithIndices(AlternatingRectangleTemplate(nsone, sheight), [2, 0]),
             # Right rectangle, containing plaquettes of type 5 only
-            TemplateWithIndices(ScalableRectangle(1, height), [0, 5]),
+            TemplateWithIndices(AlternatingRectangleTemplate(nsone, sheight), [0, 5]),
             # Bottom rectangle, containing plaquettes of type 6 only
-            TemplateWithIndices(ScalableRectangle(width, 1), [6, 0]),
+            TemplateWithIndices(AlternatingRectangleTemplate(swidth, nsone), [6, 0]),
         ]
         _relations = [
             (1, ABOVE_OF, 0),
             (2, LEFT_OF, 0),
             (3, RIGHT_OF, 0),
             (4, BELOW_OF, 0),
         ]
-        TemplateOrchestrator.__init__(self, _templates)
+        super().__init__(_templates)
         for source, relpos, target in _relations:
             self.add_relation(source, relpos, target)
+
+    @staticmethod
+    def get_dimensions(
```

Comment:
This is kind of messy to me. Maybe we can enforce the user to use `Dimension`?

Reply:
Yes, I agree. I wanted to keep backward compatibility for a while, but it is way easier to simply enforce `Dimension` everywhere.